homeassistant/components/generic_thermostat/climate.py

```diff
@@ -112,7 +112,8 @@
             vol.In([PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE]),
         ),
         vol.Optional(CONF_TEMP_STEP): vol.All(
-            vol.In([PRECISION_TENTHS, PRECISION_HALVES, PRECISION_WHOLE])
+            vol.Coerce(float),
```

Comment:
Why a float? like we support. 0.1, 0.5, 1 steps, what step is missing you like to expose this way?

Reply:
The documentation for the YAML configuration lists `precision` and `target_temp_step` as floats: https://www.home-assistant.io/integrations/generic_thermostat/#precision

The config schema for `precision` already includes the `vol.Coerce(float)` call: https://github.com/home-assistant/core/blob/779fb061f54de0820f4a962ceeec46fbc9269241/homeassistant/components/generic_thermostat/climate.py#L111

I was adding the equivalent to `target_temp_step` for consistency.